test/suite/test_dump02.py

```diff
@@ -0,0 +1,102 @@
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# [TEST_TAGS]
+# wt_util
+# [END_TAGS]
+
+import wttest
+from suite_subprocess import suite_subprocess
+
+# test_dump02.py
+# Test the dump utility to find different keys.
+class test_dump(wttest.WiredTigerTestCase, suite_subprocess):
+    table_format = 'key_format=u,value_format=u'
+    uri = 'table:test_dump'
+    output = 'dump.out'
+    data_header = 'Data\n'
+
+    n_rows = 100
+
+    def gen_key(self, i):
+        return 'key' + str(i)
+
+    def gen_value(self, i):
+        return 'value' + str(i)
+    
+    def get_num_data_lines_from_dump(self, file):
+        """
+        Get the number of lines corresponding to data from a dump file.
+        """
+        lines = open(file).readlines()
+        data_started = False
+        num_lines = 0
+        for line in lines:
+            if data_started:
+                num_lines += 1
+
+            if line == self.data_header:
+                assert not data_started
+                data_started = True
+        return num_lines
+
+    def populate_table(self, uri, n_rows):
+        cursor = self.session.open_cursor(uri, None, None)
+        for i in range(1, n_rows):
+            cursor[self.gen_key(i)] = self.gen_value(i)
+        cursor.close()
+
+    def test_dump(self):
+        self.session.create(self.uri, self.table_format)
+        self.populate_table(self.uri, self.n_rows)
+
+        self.runWt(['dump', self.uri], outfilename=self.output)
+        assert self.get_num_data_lines_from_dump(self.output) == (self.n_rows - 1) * 2
+
+
+    def test_dump_single_key(self):
+        self.session.create(self.uri, self.table_format)
+        self.populate_table(self.uri, self.n_rows)
+
+        # The key does not exist.
+        self.runWt(['dump', '-k', 'key0', self.uri], outfilename=self.output)
+        num_lines = self.get_num_data_lines_from_dump(self.output)
+        assert num_lines == 0, num_lines
+
+        # The nearest key should be found.
+        self.runWt(['dump', '-k', 'key0', '-n', self.uri], outfilename=self.output)
+        num_lines = self.get_num_data_lines_from_dump(self.output)
+        assert num_lines == 2, num_lines
```

Comment:
Question: why are 2 lines returned for one key? Is that just a function of the output format?

Reply:
Yes, find below a sample of the output:
```
WiredTiger Dump (WiredTiger Version 11.2.0)
Format=print
Header
table:test_dump
access_pattern_hint=none,allocation_size=4KB,app_metadata=,assert=(commit_timestamp=...
Data
key1
value1
key10
value10
key11
value11
key12
value12
key13
value13
key14
value14
```